src/StateManager.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+function updateBrowserHistoryStack(currentStepName, nextStepName) {
+  const currentUrl = window.location.href;
+  const basePath = currentUrl.includes('?')
+    ? currentUrl.split('?')[0]
+    : currentUrl;
+  const searchParams = new URLSearchParams(new URL(currentUrl).searchParams);
+  searchParams.set('step', nextStepName);
+
+  window.history.pushState({ activeStep: currentStepName }, '', currentUrl);
+  window.history.replaceState(
+    { activeStep: nextStepName },
+    '',
+    `${basePath}?${searchParams.toString()}`,
+  );
+}
+
+function getActiveStepFromUrl() {
+  const { href } = window.location;
+  const searchParams = new URLSearchParams(new URL(href).searchParams);
+  return searchParams.get('step');
+}
+
+export default class StateManager {
+  mode;
+
+  constructor(mode) {
+    const supportedModes = ['url'];
+
+    if (supportedModes.includes(mode)) {
+      this.mode = mode;
+    } else if (!mode) {
+      this.mode = false;
```

Comment:
I think it's confusing that `this.mode` can be both boolean and a string. Why not always throw if the `mode` isn't supported?

Reply:
Agreed, fixed in 9adb3e7